From: Lukas Nykryn <lnykryn@redhat.com>
Date: Mon, 24 Feb 2014 17:06:21 +0100
Subject: udev: properly detect reference to unexisting part of PROGRAM's
 result

(cherry picked from commit 66390abefae6530981446970ffd7ff4d1085e254)
(cherry picked from commit e97bfdb85116454e10255a55a7763415577078e2)
---
 src/udev/udev-event.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/udev/udev-event.c b/src/udev/udev-event.c
index 3db2cb7..37b545b 100644
--- a/src/udev/udev-event.c
+++ b/src/udev/udev-event.c
@@ -254,6 +254,8 @@ subst:
                                                 cpos++;
                                         while (isspace(cpos[0]))
                                                 cpos++;
+                                        if (cpos[0] == '\0')
+                                                break;
                                 }
                                 if (i > 0) {
                                         log_error("requested part of result string not found\n");
